execution_engine/src/core/execution/address_generator.rs

```diff
@@ -28,6 +28,21 @@ impl AddressGenerator {
         self.0.fill_bytes(&mut buff);
         buff
     }
+
+    pub fn new_hash_address(&mut self) -> Address {
```

Comment:
Could you leave a TODO here to investigate this later?  

I accept that it's a effectively a vestigial construction, but it's really wonky.

Reply:
did this